src/super_gradients/training/losses/yolox_loss.py

```diff
@@ -0,0 +1,468 @@
+                                   CPU mode is applied in this batch. If you want to avoid this issue, \
+                                   try to reduce the batch size or image size.")
+                    torch.cuda.empty_cache()
+                    gt_matched_classes, fg_mask, pred_ious_this_matching, matched_gt_inds, num_fg_img = \
+                        self.get_assignments(batch_idx, num_gt, total_num_anchors, gt_bboxes_per_image,
+                                             gt_classes, bboxes_preds_per_image,
+                                             expanded_strides, x_shifts, y_shifts, cls_preds, obj_preds, 'cpu')
+
+                torch.cuda.empty_cache()
+                num_fg += num_fg_img
+
+                cls_target = F.one_hot(gt_matched_classes.to(torch.int64), self.num_classes) * \
+                             pred_ious_this_matching.unsqueeze(-1)
+                obj_target = fg_mask.unsqueeze(-1)
+                reg_target = gt_bboxes_per_image[matched_gt_inds]
+                if self.use_l1:
+                    l1_target = self.get_l1_target(transformed_outputs.new_zeros((num_fg_img, 4)),
+                                                   gt_bboxes_per_image[matched_gt_inds], expanded_strides[0][fg_mask],
+                                                   x_shifts=x_shifts[0][fg_mask], y_shifts=y_shifts[0][fg_mask])
+
+            # collect targets for all loss terms over the whole batch
+            cls_targets.append(cls_target)
+            reg_targets.append(reg_target)
+            obj_targets.append(obj_target.to(transformed_outputs.dtype))
+            fg_masks.append(fg_mask)
+            if self.use_l1:
+                l1_targets.append(l1_target)
+
+        # concat all targets over the batch (get rid of batch dim)
+        cls_targets = torch.cat(cls_targets, 0)
+        reg_targets = torch.cat(reg_targets, 0)
+        obj_targets = torch.cat(obj_targets, 0)
+        fg_masks = torch.cat(fg_masks, 0)
+        if self.use_l1:
+            l1_targets = torch.cat(l1_targets, 0)
+
+        num_fg = max(num_fg, 1)
+        # loss terms divided by the total number of foregrounds
+        loss_iou = self.iou_loss(bbox_preds.view(-1, 4)[fg_masks], reg_targets).sum() / num_fg
+        loss_obj = self.bcewithlog_loss(obj_preds.view(-1, 1), obj_targets).sum() / num_fg
+        loss_cls = self.bcewithlog_loss(cls_preds.view(-1, self.num_classes)[fg_masks], cls_targets).sum() / num_fg
+        if self.use_l1:
+            loss_l1 = self.l1_loss(raw_outputs.view(-1, 4)[fg_masks], l1_targets).sum() / num_fg
+        else:
+            loss_l1 = 0.0
+
+        reg_weight = 5.0
+        loss = reg_weight * loss_iou + loss_obj + loss_cls + loss_l1
+
+        return loss, torch.cat((loss_iou.unsqueeze(0), loss_obj.unsqueeze(0), loss_cls.unsqueeze(0),
+                                torch.tensor(loss_l1).unsqueeze(0).cuda(),
```

Comment:
do not use .cuda() 
use to(device) instead

Reply:
done